packages/blocks/src/phone-number/index.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { RawHTML } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import classnames from 'classnames';
+import { isEmpty } from 'lodash';
+import 'react-phone-number-input/style.css';
+import PhoneInput from 'react-phone-number-input';
+import Input from 'react-phone-number-input/input';
+
+/**
+ * Internal dependencies
+ */
+import { useColorStyles } from '@crowdsignal/styles';
+import { ErrorMessage, FormInputWrapper } from '../components';
+import { useField } from '@crowdsignal/form';
+
+const PhoneNumber = ( { attributes, className } ) => {
+	const flag = attributes.flag;
+	const { error, onChange, fieldValue } = useField( {
+		fieldName: `q_${ attributes.clientId }[text]`,
+		validation: ( value ) => {
+			if ( attributes.mandatory && isEmpty( value ) ) {
+				return __( 'This field is required', 'blocks' );
+			}
+		},
+	} );
+
+	const classes = classnames(
+		className,
+		'crowdsignal-forms-phone-number-block',
+		{
+			'is-required': attributes.mandatory,
+			'is-error': error,
+		}
+	);
+
+	return (
+		<FormInputWrapper
+			className={ classes }
+			style={ { ...useColorStyles( attributes ) } }
+		>
+			<FormInputWrapper.Label className="crowdsignal-forms-phone-number-block__label">
+				<RawHTML>{ attributes.label }</RawHTML>
+			</FormInputWrapper.Label>
+			{ flag && (
+				<PhoneInput
+					defaultCountry={ attributes.country.toUpperCase() }
+					onChange={ onChange }
+					placeholder={ attributes.placeholder }
+					style={ {
+						width: attributes.inputWidth,
+					} }
+					value={ fieldValue }
+				/>
+			) }
+			{ ! flag && (
+				<Input
+					defaultCountry={ attributes.country.toUpperCase() }
+					onChange={ onChange }
+					placeholder={ attributes.placeholder }
+					style={ {
+						width: attributes.inputWidth,
+					} }
+					value={ fieldValue }
+				/>
+			) }
```

Comment:
Is there a more elegant way to write this? the `<PhoneInput>` is the version with the flag and `<Input>` is the non-flagged version.

Reply:
Yes. You could move this piece of code to a new React component. 
A short way would be something like this:

```JavaScript
	const PhoneBlock = ( props ) => ( flag ? <PhoneInput { ...props } /> : <Input { ...props } /> );
	
	//Now you could do this:
	<PhoneBlock
		defaultCountry={ attributes.country.toUpperCase() }
		placeholder={ __(
			'Enter Phone Number',
			'block-editor'
		) }
		value={ attributes.placeholder }
		onChange={ noop }
		className="crowdsignal-forms-phone-number-block__wrapper"
	/>
```
Here I'm getting the `flag` value from the outer context, the main component. 
But you could accept the flag as a prop and pass it when instantiating the component.

However, if we do that, we'd still end up with this logic duplicated for the editor and the renderer.
We can avoid that by creating a new component inside `packages/blocks/components/form-phone-numer`:

```JavaScript
/**
 * External dependencies
 */
import PhoneInput from 'react-phone-number-input';
import Input from 'react-phone-number-input/input';

const FormPhoneNumber = ( { flag, ...props } ) => {
	if ( flag ) {
		return <PhoneInput { ...props } />
	}

	return <Input { ...props } />
}

export default FormPhoneNumber

```

Make sure to export it in the `index.js` file.
And now, you can use it in the editor and in the renderer:

```JavaScript
	<FormPhoneNumber
		flag={ flag }
		...
	/>
```

Reply:
Okay, perfect. This makes a lot of sense. Thanks!